### PR TITLE
Simplenote: init at 1.0.6

### PR DESCRIPTION
**Pull request by dyrnade**
###### Motivation for this change
Simplenote Package

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/simplenote/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ stdenv.mkDerivation rec {
+
+  name = "simplenote-${pkgver}";
+  pkgver = "1.0.6";
+
+  src = fetchurl {
+    url = "https://github.com/Automattic/simplenote-electron/releases/download/v${pkgver}/${name}.deb";
+    md5 = "752fa5f16e22f3a8e5cc4a87195099de";
+    name = "${name}.deb";
+  };
```
Comment (kevincox) [inline pkgs/applications/misc/simplenote/default.nix]: We strongly prefer building from source. Is this possible?

**Review by kevincox — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/simplenote/default.nix (hunk 1 above)
Comment (dyrnade) [inline pkgs/applications/misc/simplenote/default.nix]: Okey.I will have a look at it.

**Comment by dyrnade** [comment]
@kevincox updated package.

**Comment by kevincox** [comment]
It appears you are still using a (different) built version. You probably want to start by using `fetchFromGithub` to clone the repository but I'm not an electron expert so hopefully @TikhonJelvis or @ejlo can give you some hints about how to package electron apps. Election is already in nixpkgs however it that we have [one other package using it](https://github.com/NixOS/nixpkgs/blob/56904d7c423f2b13b37fbd29f39bbb4b52bc7824/pkgs/applications/misc/nix-tour/default.nix) so hopefully that can help as a guide..

**Comment by dyrnade** [comment]
@kevincox this is the best possible way to install simplenote on nix. I looked at installations on archlinux for couple of electron apps, all of them use deb or similar ways.

Slack installation uses also deb package.
https://github.com/NixOS/nixpkgs/blob/7098404505003887b62cdee10ee74dfa56ead741/pkgs/applications/networking/instant-messengers/slack/default.nix

From instructions here https://github.com/Automattic/simplenote-electron

npm run build doesnt work.

**Comment by kevincox** [comment]
You might be right but I would like to have the electron maintainers opinion first. Not only because this bundles it's own copy of electron which is probably uncessary but also because source builds are preferred. 

Slack is unfortunately closed source so binary is the only option. 

I get the simplenote recommends the binary to end users but most distributions will repackage. I think it would be best to do that if possible.

**Comment by dyrnade** [comment]
@kevincox Yeap, you are right, maybe there is a better solution. As far as i ve seen, all use deb packages (atom too).

**Review by kevincox — APPROVED** [review]

**Comment by kevincox** [comment]
Well if there is prescient I guess I'm ok with it if the electron guys are. Maybe it'll inspire me to create a framework for properly packaging electron apps.

**Comment by FRidh** [comment]
commits needs to be squashed and named according the guidelines

**Comment by dyrnade** [comment]
@FRidh squashed commits.

**Comment by dyrnade** [comment]
ping @FRidh @kevincox

**Pull request merged**